src/protocol/mod.rs

```diff
@@ -0,0 +1,53 @@
+use crate::packet::{HandshakeControlInfo, ShakeType};
```

Comment:
I haven't been using `mod.rs` files in this project, preferring a `protocol.rs` in the parent directory. I don't really care either way but for consistency I would like it to be `protocol.rs`.

Same goes for `tokio/mod.rs`